static/js/services/relative-date.js

```diff
@@ -0,0 +1,63 @@
+var moment = require('moment');
+
+angular.module('inboxServices').factory('RelativeDate',
+  function(
+    FormatDate
+  ) {
+    'use strict';
+    'ngInject';
+
+    var config = {
+      cssSelector: 'update-relative-date'
+    };
+
+    var getDataAttribute = function(key, value) {
+      return 'data-' + camelCaseToDash(key) + '="' + value + '"';
+    };
+
+    var camelCaseToDash = function(string) {
+      return string.replace(/[A-Z]/g, function(match) {
+        return '-' + match[0].toLowerCase();
+      });
+    };
+
+    var skipOptions = ['FormatDate', 'RelativeDate', 'suffix', 'prefix'];
+
+    return {
+      getCssSelector: function() {
+        return config.cssSelector;
+      },
+      generateDataset: function(date, options) {
+        var dataAttributes = [];
+        var momentDate = moment(date);
+        dataAttributes.push(getDataAttribute('date', momentDate.valueOf()));
+
+        for (var key in options) {
+          if (typeof options[key] !== 'object' && skipOptions.indexOf(key) === -1 && options[key]) {
+            dataAttributes.push(getDataAttribute(key, options[key]));
+          }
+        }
+
+        return dataAttributes.join(' ');
+      },
```

Comment:
I think it'd be cleaner to store the options in an object on a single data attribute rather than each in their own, eg:

```html
<span data-date-options="{ withoutTime: true, ... }">
```

This means you don't need to split the options up, kebab case the names, and then reverse the process when reading them again.

Reply:
aaand done.